native/cocos/editor-support/spine-wasm/spine-wasm.cpp

```diff
@@ -4,10 +4,57 @@
 #include "util-function.h"
 #include "wasmSpineExtension.h"
 
+#include <emscripten/emscripten.h>
+#include <emscripten/val.h>
+
+#include "spine/HashMap.h"
+
 using namespace spine;
 
+static void logToConsole(const char* message) {
+    EM_ASM({
+        console.log(UTF8ToString($0));
+    }, message);
+}
+
 namespace {
     HashMap<String, SkeletonData*> skeletonDataMap{};
+
+    static void updateAttachmentVerticesTextureId(SkeletonData* skeletonData, const spine::Vector<spine::String>& textureNames, const spine::Vector<spine::String>& textureUUIDs) {
+        spine::HashMap<spine::String, spine::String> textureMap{};
+        int textureSize = textureNames.size();
+        for (int i = 0; i < textureSize; ++i) {
+            textureMap.put(textureNames[i], textureUUIDs[i]);
+        }
+
+        auto& skins = skeletonData->getSkins();
+        auto skinSize = skins.size();
+        for (int i = 0; i < skinSize; ++i) {
+            auto* skin = skins[i];
+            auto entries = skin->getAttachments();
+            while (entries.hasNext()) {
+                Skin::AttachmentMap::Entry& entry = entries.next();
+                AttachmentVertices* attachmentVertices;
+                auto* attachment = entry._attachment;
+                if (attachment->getRTTI().isExactly(MeshAttachment::rtti)) {
+                    auto* meshAttachment = static_cast<MeshAttachment *>(attachment);
+                    attachmentVertices = static_cast<AttachmentVertices*>(meshAttachment->getRendererObject());
+                } else {
+                    auto* regionAttachment = static_cast<RegionAttachment *>(attachment);
+                    attachmentVertices = static_cast<AttachmentVertices*>(regionAttachment->getRendererObject());
+                }
+                auto& textureName = attachmentVertices->_textureId;
+                if (textureMap.containsKey(textureName)) {
+                    attachmentVertices->_textureId = textureMap[textureName];
+                } else {
+                    spine::String logInfo(attachment->getName());
+                    logInfo.append(" attachment's texture is not exist ");
+                    logInfo.append(textureName);
+                    logToConsole(logInfo.buffer());
```

Comment:
Why not use `printf` directly?

Reply:
I got it. `printf` will make the wasm size 500 bytes bigger. So don't use `printf` is a good choice.